### ISSUE TITLE
pastec "Could not open the backward index file"

### ISSUE THREAD

**Issue by bixmatech** [magwyz/pastec#72]
Hi,
pastec "Could not open the backward index file", so each time vps restart, indexes gone, anyhelp please?

**Comment by magwyz** [comment]
You need to manually use the WRITE command.
The index is not automatically saved.

**Comment by bixmatech** [comment]
but what if we say the pastec have about new 10k images indexed and the vps crashed and rebooted, in this case all these images will be lost!

**Comment by abournier** [comment]
Create a cron to write the index all X minutes.

**Comment by magwyz** [comment]
Add your 10k images and call the WRITE command.
I know this is annoying but Pastec custom database is basic and does not support incremental disk write.

**Comment by abournier** [comment]
Every hour example:

`0 * * * * curl -X POST -d '{"type":"WRITE"}' http://127.0.0.1:4212/index/io > /home/admin/pastec/build/cron.log 2>&1`

**Comment by abournier** [comment]
> Add your 10k images and call the WRITE command.
> I know this is annoying but Pastec custom database is basic and does not support incremental disk write.

Writing the index in a real database like PGSQL could be possible with lots of work,  or is not possible at all due to Pastec image recognition engine ?

**Comment by bixmatech** [comment]
yeah i was thinking about to save the index by calling write every hour per example, however, if the vps is restarted in this hour for a failure per example, this will ignore added images in this time between last write command and the failure date, which make pastec unrialable solution for use in production, got me or i am wrong?
I was thinking about merging the save index with the add image function, or invoke the write command whatever image is  added but this will do alot of stress to the vps

**Comment by magwyz** [comment]
> yeah i was thinking about to save the index by calling write every hour per example, however, if the vps is restarted in this hour for a failure per example, this will ignore added images in this time between last write command and the failure date, which make pastec unrialable solution for use in production, got me or i am wrong?

Pastec may be unreliable in production for YOUR use case. However, please do not make your case a generality. Some users do not require to permanently update the database.

Besides, instead of criticizing the project, you could also just handle yourself the recovery after an incident. Save on disk all the images that have been added to the index since your last WRITE call and reconstruct the index with the missing images on recovery.

> I was thinking about merging the save index with the add image function, or invoke the write command whatever image is added but this will do alot of stress to the vps

I would not advise you to do that since the WRITE command write all the index in a file. This is very costly, especially if you have a lot of images in your index.

**Comment by bixmatech** [comment]
You are right, so what do you suggest to keep the data backed up and auto saved in index inc ase of vps failure and restart?

**Comment by abournier** [comment]
Maybe you can save the datetime in a DB when an image is added ot the index. If you get the index save time, you can compare to images datetime and add the images lost in the index once again.

**Comment by bixmatech** [comment]
@magwyz :

How much it costs if we want you to dd these 2 features;
1- autosave new images to indexes in case of system failure and reboot
2- Make pastec dont add the same image twice to it, so in case if it is found, it return a feedback that this image is found with it's id?

@magwyz @magwyz :
1- what is the name of the default index that pastec uses when it run?
2- I saved the index test.dat and able to load it, but what is the path of the .dat file? as I didnt find it near the pastec assembly in case i want to move it to another place?
3- I saved an index with name test.dat, and i used the argument -i test.dat to load it with pastec lanch but it didnt loaded!

@magwyz :
Yes this is what I am thinking of
